include/aunit/framework/nocalendar/aunit-io.ads

```diff
@@ -0,0 +1,52 @@
+------------------------------------------------------------------------------
+--                                                                          --
+--                         GNAT COMPILER COMPONENTS                         --
+--                                                                          --
+--                        A U N I T . R E P O R T E R                       --
+--                                                                          --
+--                                 S p e c                                  --
+--                                                                          --
+--                                                                          --
+--                       Copyright (C) 2019, AdaCore                        --
+--                                                                          --
+-- GNAT is free software;  you can  redistribute it  and/or modify it under --
+-- terms of the  GNU General Public License as published  by the Free Soft- --
+-- ware  Foundation;  either version 3,  or (at your option) any later ver- --
+-- sion.  GNAT is distributed in the hope that it will be useful, but WITH- --
+-- OUT ANY WARRANTY;  without even the  implied warranty of MERCHANTABILITY --
+-- or FITNESS FOR A PARTICULAR PURPOSE.                                     --
+--                                                                          --
+-- As a special exception under Section 7 of GPL version 3, you are granted --
+-- additional permissions described in the GCC Runtime Library Exception,   --
+-- version 3.1, as published by the Free Software Foundation.               --
+--                                                                          --
+-- You should have received a copy of the GNU General Public License and    --
+-- a copy of the GCC Runtime Library Exception along with this program;     --
+-- see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see    --
+-- <http://www.gnu.org/licenses/>.                                          --
+--                                                                          --
+-- GNAT is maintained by AdaCore (http://www.adacore.com)                   --
+--                                                                          --
+------------------------------------------------------------------------------
+
+with GNAT.IO;
+
+package AUnit.IO is
+
+   subtype File_Type is GNAT.IO.File_Type;
+
+   type File_Access is access constant GNAT.IO.File_Type;
```

Comment:
Unfortunately that still won't work as in restricted runtimes there is no File_Type at all. GNAT.IO basically looks like this:

```
package GNAT.IO is
   pragma Preelaborate;

   procedure Put (X : Integer);
   --  Output integer to specified file, or to current output file, same
   --  output as if Ada.Text_IO.Integer_IO had been instantiated for Integer.

   procedure Put (C : Character);
   --  Output character to specified file, or to current output file

   procedure Put (S : String);
   --  Output string to specified file, or to current output file

   procedure Put_Line (S : String);
   --  Output string followed by new line to specified file, or to
   --  current output file.

   procedure New_Line (Spacing : Positive := 1);
   --  Output new line character to specified file, or to current output file

end GNAT.IO;```

Reply:
Ok, then I will implement File_Type/File_Access as dummy types for the restricted runtimes. Is there a way I can test this in advance?

Reply:
That should work I think.
If you do not have acces to a cross toolchain, you can download gnat community edition for arm-elf target from AdaCore website, put it in path and then attempt to build aunit for any zfp runtime, for example like this:
```make TARGET=arm-eabi RTS=zfp RTS_CONF="--RTS=zfp-stm32f4"```

Reply:
Have a look at my latest commit, also checked this with the ARM cross compiler.